frontend/src/metabase/writeback/components/ActionCreator/FormCreator/FormCreator.tsx

```diff
@@ -39,6 +52,13 @@ export function FormCreator({
     onChange(formSettings);
   }, [formSettings, onChange]);
 
+  useEffect(() => {
+    // add default settings for new parameters
+    if (formSettings && params && hasNewParams(params, formSettings)) {
+      setFormSettings(addMissingSettings(formSettings, params));
+    }
+  }, [params, formSettings]);
```

Comment:
previously, we wouldn't add field settings until they opened the field settings popover. We need those settings to save order though, so this adds default settings any time we get a parameter that doesn't have any settings yet.

Reply:
Should we consider adding new field settings when a new parameter is added?
IMHO, `useEffect` makes this logic a bit entangled

Reply:
That's effectively what we're doing here - we don't have an event listener or anything for when a parameter gets added, it's just when we get passed a new template tag from `question.parameters()`

I agree it's not the clearest bit of logic. I'd be interested if you had any ideas for making it clearer?

Reply:
Yeah, I was hoping there's a prop like `onParametersChange` or something that'd let us make this logic a bit more straightforward, but seems like `useEffect` is our best shot at the moment. For the future, is the native query editor component a good place to expose this kind of a prop? WDYT?